lib/libzfs/libzfs_sendrecv.c

```diff
@@ -4638,6 +4638,33 @@ zfs_receive_one(libzfs_handle_t *hdl, int infd, const char *tosnap,
 			goto out;
 		}
 
+		/*
+		 * When receiving full/newfs on existing dataset, then it
+		 * should be done with "-F" flag. Its enforced for initial
+		 * receive in previous checks in this function.
+		 * Similarly, on resuming full/newfs recv on existing dataset,
+		 * it should be done with "-F" flag.
+		 *
+		 * When dataset doesn't exist, then full/newfs recv is done on
+		 * newly created dataset and it's marked INCONSISTENT. But
+		 * When receiving on existing dataset, recv is first done on
+		 * %recv and its marked INCONSISTENT. Existing dataset is not
+		 * marked INCONSISTENT.
+		 * Resume of full/newfs receive with dataset not INCONSISTENT
+		 * indicates that its resuming newfs on existing dataset. So,
+		 * enforce "-F" flag in this case.
+		 */
+		if (stream_resumingnewfs &&
+		    !zfs_prop_get_int(zhp, ZFS_PROP_INCONSISTENT) &&
+		    !flags->force) {
+			zfs_close(zhp);
+			zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
+			    "destination '%s' exists\n"
+			    "must specify -F to overwrite it"), name);
+			err = zfs_error(hdl, EZFS_EXISTS, errbuf);
+			goto out;
+		}
```

Comment:
This check needs to be done in the kernel, otherwise it'll still be possible to panic the system in this way.  You can add a new `zfs_errno_t` for this case.

Reply:
Ok.  For initial receive, Kernel does return "EEXIST" error  via dmu_recv_being_check()->recv_begin_check_existing_impl() , when full/newfs recv is done on Existing dataset without force flag.
I would add a similar check in dmu_recv_resume_begin_check() to return EEXIST error.

I would also check about new zfs_errno_t and message.

Reply:
Added new error code "ZFS_ERR_RESUME_EXISTS" in zfs_errno_t and returning it from kernel for this case.
And handling in libzfs for this error code.